buildSrc/src/main/kotlin/org/gradle/plugins/compile/AvailableJavaInstallations.kt

```diff
@@ -26,12 +30,33 @@ class DefaultJavaInstallation(val current: Boolean, private val javaHome: File)
     override fun toString(): String = "${displayName} (${javaHome.absolutePath})"
 }
 
-open class AvailableJavaInstallations(javaHomesForCompilation: List<String>, javaHomeForTest: String?, private val javaInstallationProbe: JavaInstallationProbe) {
-    private val javaInstallations: Map<JavaVersion, DefaultJavaInstallation> = findJavaInstallations(javaHomesForCompilation)
+open class AvailableJavaInstallations(project: Project, private val javaInstallationProbe: JavaInstallationProbe) {
+
+    private
+    companion object {
+        private
+        val java7HomePropertyName = "java7Home"
+        private
+        val testJavaHomePropertyName = "testJavaHome"
+        private
+        val oracleJdk8 = "Oracle JDK 8"
+        private
+        val oracleJdk7 = "Oracle JDK 7"
```

Comment:
:o: This could be simplified by using top level consts, e.g.:

```kotlin
private
const val java7HomePropertyName = "java7Home"
```

Reply:
Done. I was not expecting to have `const`, too, given that `val` is already immutable...